src/core/lender/assessor/assessor.sol

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2019 Centrifuge
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.5.12;
+
+import "./base.sol";
+
+contract Assessor is BaseAssessor {
+
+    // accrueTrancheInterest can implement different interest models
+    function accrueTrancheInterest(address tranche_) public returns (uint) {
+        SeniorTrancheLike tranche = SeniorTrancheLike(tranche_);
+        uint debt = tranche.debt();
```

Comment:
`assessor.accrueTrancheInterest` calls`tranche.debt()` which calls `drip()` which calls `assessor.accrueTrancheInterest()` 🙈

Reply:
True. 👍  It should be `uint debt = `tranche.borrowed()+tranche.interest()`.